### PR TITLE
Couch db

### PR DESCRIPTION
**Pull request by jeremylfchen**
Hi @Nick-Zabalza @cmonez ,

Here is my PR for seeding couchDB. Please excuse the super hacky way for my data generation, this was done so I didn't have to rewrite my csv generating files lol 

All it does is parse the stores csv once to get the 5 store infos, and parse the availability csv every 5 lines to create 1 item entry. After the batch size has been met, I put the read stream on pause and send a post request to write the data into the database. After a 500ms delay, I resume the process all over again until i reach the end of the availability csv.

Let me know if you have any questions!

### CODE REVIEW

**Comment by cmonez** [comment]
> Hi @Nick-Zabalza @cmonez ,
> 
> Here is my PR for seeding couchDB. Please excuse the super hacky way for my data generation, this was done so I didn't have to rewrite my csv generating files lol
> 
> All it does is parse the stores csv once to get the 5 store infos, and parse the availability csv every 5 lines to create 1 item entry. After the batch size has been met, I put the read stream on pause and send a post request to write the data into the database. After a 500ms delay, I resume the process all over again until i reach the end of the availability csv.
> 
> Let me know if you have any questions!

ill look at this rn

File: database-couchdb/index.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
+const fs = require('fs');
+const PATH = require('path');
+const csvParser = require('csv-parser');
+const Axios = require('axios');
+
+const availabilityPath = PATH.resolve(__dirname, '..', 'data-gen', 'csv', 'availability.csv');
+const storesPath = PATH.resolve(__dirname, '..', 'data-gen', 'csv', 'stores.csv');
+
+const startingItemId = 100;
+const batchSize = 100;
```
Comment (cmonez) [inline database-couchdb/index.js:10]: batchSize means ur uploading documents to couch db 100 at a time? I read you could bulk insert 100,000 at a time? Why'd you settle on 100? How long did seeding take? More for my own curiosity

File: database-couchdb/index.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    let curId = curIds
+    availabilityStream
+      .on('error', (err) => {
+        console.error(err);
+      })
+      .pipe(csvParser())
+      .on('data', (data) => {
+
+        count++;
+        if (entry.itemId === undefined) {
```
Comment (cmonez) [inline database-couchdb/index.js:37]: What does this mean exactly?

File: database-couchdb/index.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        }
+        if (docs.length === batchSize) {
+
+          console.log(curId);
+          availabilityStream.pause();
+
+          Axios.post(url + db + '/_bulk_docs', { docs })
+            .then(() => {
+              console.log('posted!');
+              setTimeout(() => {
```
Comment (cmonez) [inline database-couchdb/index.js:57]: Whats the point of having this setTimeOut waiting 5 seconds?

File: database-couchdb/index.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    .pipe(csvParser())
+    .on('data', (data) => {
+      let store = data;
+      store.storeId = data.store_id;
+      store.availability = false;
+      // console.log(store);
+      stores.push(store);
+    })
+    .on('end', () => {
+      storesStream.destroy();
```
Comment (cmonez) [inline database-couchdb/index.js:96]: Whats the point of destroying the stream?

**Review by cmonez — APPROVED** [review]
So this is reading your Postgres seeding files, correct? It seems like a very clever way, and you look so comfortable with streams. I had just a couple clarifying questions. We seeded totally different and I really like your approach. Very clever and efficient for memory.

File: database-couchdb/index.js (hunk 2 above)
Comment (jeremylfchen) [reply database-couchdb/index.js:37]: Basically after every 5 lines from the csv, i reset `entry` to be an empty object. If the key itemId doesnt exist, that means entry is ready to be repopulated with new information

File: database-couchdb/index.js (hunk 1 above)
Comment (jeremylfchen) [reply database-couchdb/index.js:10]: LOL this was through hella trial and error. I was overloading my laptop when my batch uploads were more than 100 at a time (even with my set timeouts being 10000 ms). For the final parameters i went with, it only took about 20-30 min i think.

I was getting a lot of 400 or 500 status codes before that. Also it took my laptop a long time to optimize/ compress the database after it was seeded, but I think thats just a couchdb thing. It was using 400% of my cpu for a few hours...

File: database-couchdb/index.js (hunk 3 above)
Comment (jeremylfchen) [reply database-couchdb/index.js:57]: I was getting 400 error codes when my set timeouts were less than that.

File: database-couchdb/index.js (hunk 4 above)
Comment (jeremylfchen) [reply database-couchdb/index.js:96]: Just to be safe lol I dont think its super necessary. Originally i wanted it to read the store csv file each time i made an entry but realized it was very inefficient

**Pull request merged**